src/shogun/mathematics/linalg/ratapprox/logdet/LogDetEstimator.cpp

```diff
@@ -148,72 +125,81 @@ SGVector<float64_t> CLogDetEstimator::sample(index_t num_estimates)
 		m_operator_log->get_operator()->get_dimension(),
 		m_trace_sampler->get_dimension());
 
-	// for storing the aggregators that submit_jobs return
-	CDynamicObjectArray* aggregators=new CDynamicObjectArray();
+	// for storing the result
+	CDynamicArray<float64_t> aggregators;
 	index_t num_trace_samples=m_trace_sampler->get_num_samples();
-
-	for (index_t i=0; i<num_estimates; ++i)
+	// for omp
```

Comment:
great that you use mp here.
Pull this out into the separate patch then I can review it smoothly